### PR TITLE
OOP school library: refactor your code

### PR DESCRIPTION
**Pull request by KingsleyIbe**
## Milestone changes

In this milestone, @imubarak234, and @KingsleyIbe implemented the following:

- Ensured that the role of main.rb is only to start your code and ask the user for the initial choice of options (or to quit).
- Extract some pieces of logic into separate classes.

### CODE REVIEW

File: book_main.rb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+class Bookmain
+  def list_all_books(books)
+    books.each { |n| puts n.to_string }
+  end
+
+  def create_a_book(books)
+    print 'Title: '
+    title = gets.chomp
+    print 'Author: '
+    author = gets.chomp
```
Comment (TedLivist) [inline book_main.rb]: - [ ] All requests for user inputs and validations should be handled by another method in keeping with the single-responsibility rule. Kindly extract these requests here into another method.

  **SUGGESTION:** You can extract these requests methods into another method and have that method return and array of title and author. Then destructure the variables in create_a_book like so:

  imuba-books

File: person_main.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    people.push(student)
+  end
+
+  def create_a_teacher(people)
+    puts 'Age: '
+    age = gets.chomp
+    puts 'Name: '
+    name = gets.chomp
+    puts 'Specilaization'
+    specialization = gets.chomp
```
Comment (TedLivist) [inline person_main.rb]: - [ ] Your requests for both `age` and `name` in the **`create_a_student`** and **`create_a_teacher`** methods here break the DRY rule. Kindly extract these requests into one method and call the method in both method instead.

- [ ] The request for `permission` and its validation should be handled by another method

- [ ] The request for `specialization` should be handled by another method.

File: rental_main.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    puts 'Select a book from the following list of books'
+    books.each_with_index { |n, idx| puts "#{idx}) #{n.to_string}" }
+    var = gets.chomp
+    book = books[var.to_i]
+    puts 'Select a person from the following list by number'
+    people.each_with_index { |n, idx| puts "#{idx}) #{n.to_string}" }
+    vars = gets.chomp
+    person = people[vars.to_i]
+    puts 'Enter a date'
+    date = gets.chomp
```
Comment (TedLivist) [inline rental_main.rb]: - [ ] Kindly ensure that these requests for user inputs are refactored into a different method.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @KingsleyIbe and @imubarak234,

Good job so far!
✔️ Classes are in different files
✔️ Nice refactoring
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
- [ ] In your README, kindly add instructions on how users can set up and run the app locally.
- [ ] This is a collaborative project but your README indicates only one person as a contributor. Kindly fix that.

  
  imuba-author

- [ ] It would be nice to clean up your app to look neater. When I run your app I get the following unnecessary print messages. Kindly fix that. This is due to the print messages in your `classroom.rb` and `rental.rb` files.

  imuba-puts

- [ ] _Check the comments under the review._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by TedLivist — APPROVED** [review]
## STATUS: APPROVED ✔️ ✔️

Hi @KingsleyIbe and @imubarak234,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. And please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: rental_main.rb (hunk 3 above)
Comment (KingsleyIbe) [reply rental_main.rb]: Thank you for your feedback, Changes will be implemented

**Pull request merged**